lib/ical/recur_iterator.js

```diff
@@ -727,12 +739,16 @@ ICAL.RecurIterator = (function() {
 
         if (day < 0) {
           day = daysInMonth + day + 1;
+          if (day <= 0) {
+            // This month doesn't have enough days for the current BYMONTHDAY.
+            data_valid = 0;
+          }
         }
 
         if (day > daysInMonth) {
           this.last.day = 1;
           data_valid = this.is_day_in_byday(this.last);
-        } else {
+        } else if (day > 0) {
           this.last.day = day;
         }
```

Comment:
Is there a possibility for an else case? If so there should be some sort of fallback.